src/devtools/client/inspector/computed/reducers/index.ts

```diff
@@ -0,0 +1,43 @@
+import { createReducer, ReducerObject } from "../../shared/reducer-object";
+import { ComputedState } from "../state";
+import { ComputedAction } from "../actions";
+
+const INITIAL_COMPUTED: ComputedState = {
+  properties: [],
+  search: "",
+  showBrowserStyles: false,
+  expandedProperties: [],
+};
+
+const reducers: ReducerObject<ComputedState, ComputedAction> = {
+  set_computed_properties(state, { properties }) {
+    return { ...state, properties };
+  },
+
+  set_computed_property_search(state, { search }) {
+    return { ...state, search };
+  },
+
+  set_show_browser_styles(state, { show }) {
+    return { ...state, showBrowserStyles: show };
+  },
+
+  set_computed_property_expanded(state, { property, expanded }) {
+    const expandedProperties = [...state.expandedProperties];
+
+    if (expanded) {
+      if (expandedProperties.indexOf(property) < 0) {
+        expandedProperties.push(property);
+      }
+    } else {
+      const index = expandedProperties.indexOf(property);
+      if (index >= 0) {
+        expandedProperties.splice(index, 1);
+      }
+    }
```

Comment:
should expanded properties be a Set?

Reply:
I have extracted the the components from `ComputedApp` and switched `expandedProperties` to `Set<string>`, as suggested. 
The main changes compared to your gist:
- I put the extracted components in separate files
- I don't connect `ComputedApp` to the redux store, only to pass down the stuff I get from the store to the new components. Instead, I connect those components to the store directly.
- I put the "No CSS properties found" div inside the `ComputedProperties` component
- instead of hiding the "No CSS properties found" div and the list of properties, you remove them from the DOM. I decided to stick to hiding them because it should be slightly better for performance and I didn't see any benefits from removing instead of hiding them.